src/typing_extensions.py

```diff
@@ -942,11 +942,6 @@ def __new__(cls, name, bases, ns, *, total=True):
                 else:
                     optional_keys.add(annotation_key)
                 if ReadOnly in qualifiers:
-                    if annotation_key in mutable_keys:
```

Comment:
Did you not want to replace this with a `mutable_keys.discard`? I think that would be symmetric with how this code handles other "illegal" overrides, e.g. with Required.